### PR TITLE
Display advice in empty editor to use Mito AI

### PR DESCRIPTION
**Pull request by fcollonval**
# Description

Display advice to trigger the chat panel in empty cells

# Testing

Work out of the box
Added an integration test for the feature

https://github.com/user-attachments/assets/a06ce49f-6784-4014-9d34-44f7f658fce5

Latest version looks like that:

image

# Documentation

N/A

### CODE REVIEW

File: mito-ai/src/Extensions/emptyCell/emptyCell.ts
Comment (aarondr77) [inline mito-ai/src/Extensions/emptyCell/emptyCell.ts:1]: A small nitpick, so if it's not an easy fix, let's ignore it for now. 

But it would be nice if in an empty cell the user's cursor was still at the very start of the cell instead after the placeholder text.

**Review by aarondr77 — COMMENTED** [review]
This is super sweet! I love this. It's a great touch.

File: mito-ai/src/Extensions/emptyCell/emptyCell.ts
Comment (fcollonval) [reply mito-ai/src/Extensions/emptyCell/emptyCell.ts:1]: Reolved in f3cc190

File: mito-ai/src/Extensions/emptyCell/emptyCell.ts
Comment (aarondr77) [inline mito-ai/src/Extensions/emptyCell/emptyCell.ts:30]: My understanding is that when there are multiple code mirror extensions in one code mirror editor, this `combine` function is used to figure out how to reconcile them. So here, we are saying something like: If there are multiple chatShorcut decorations set, use the last one. 

Is this needed in order to update the decorations displayed? ie: If instead of returning the last value, we instead returned the first one, would we always display the placeholder text?

File: mito-ai/ui-tests/tests/mitoAdvice.spec.ts
Comment (aarondr77) [inline mito-ai/ui-tests/tests/mitoAdvice.spec.ts:1]: Just leaving this as a note that we decided we're going to move this test into the `tests` folder at the root of the monorepo. This test will then automatically run whenever we edit the `mito-ai` folder.

File: mito-ai/ui-tests/tests/mitoAdvice.spec.ts
Comment (aarondr77) [inline mito-ai/ui-tests/tests/mitoAdvice.spec.ts:17]: ooo cool! I didn't know this soft assertion was a thing. This is great for letting us test each of the different permutations in one test. I like it.

**Review by aarondr77 — COMMENTED** [review]
Just a little bit of cleanup and then LGTM without another review. Nice work! The tests aren't going to pass on a forked PR, so we'll see them pass on the CI after merging. And in the future, now that you have write access, you can open cloned PRs and the tests should run + pass there :)

File: mito-ai/src/Extensions/emptyCell/emptyCell.ts
Comment (fcollonval) [reply mito-ai/src/Extensions/emptyCell/emptyCell.ts:49]: First part was a bug due to this editor not being force to be loaded after the AI chat plugin. Therefore the shortcut was not yet defined and it fall back to the default 'Ctrl + E'.

I updated the code to for plugin load order via token dependency and changed the default to 'Accel + E'.

File: mito-ai/src/Extensions/emptyCell/emptyCell.ts
Comment (fcollonval) [reply mito-ai/src/Extensions/emptyCell/emptyCell.ts:30]: Actually this apply only for the `Facet` that is storing configuration information. Using the latest value is usually the best option if we don't know better.

> It is unlikely to happen in this case.

File: mito-ai/ui-tests/tests/mitoAdvice.spec.ts
Comment (fcollonval) [reply mito-ai/ui-tests/tests/mitoAdvice.spec.ts:17]: Yep and it is recommended as best practice: https://playwright.dev/docs/best-practices#use-soft-assertions

**Review by aarondr77 — APPROVED** [review]

**Pull request merged**
